src/wasm-type.h

```diff
@@ -28,6 +28,7 @@ enum Type {
   f32,
   f64,
   v128,
+  anyref,
```

Comment:
does this need to be the last leg of the enum so the values of the existing legs don't shift or non-issue?

Reply:
I don't think so..? Why should it matter?

Reply:
It might matter to someone using the C API in whatever language and they create their own enum with these values instead of using the `BinaryenTypeInt32()` functions since some languages (e.g. C++) can't switch through non-constants.

Wasn't sure how careful binaryen tries to be with breaking things for folks. I imagine it's not worth the effort at this point, the number of compilers using Binaryen I assume isn't massive.

Reply:
I think people shouldn't rely on the values of enums. But just in case if you are worried about it, you can append `anyref` at the end of the list I think, rather than inserting it in the middle..?

Reply:
Friendly ping

Reply:
oops sorry. I don't care enough to move it because I'm pretty confident it will happen many many many times in the future and won't be thought about because it's super easy to not realize might be depending on it implicitly. I'd rather have the enum arm order make more logical sense internally. (in this case, exnref being after anyref makes more sense)

I mostly wanted to point it out in case you or the other maintainers already had decided to make this a stable, public API.

Reply:
I did mention my use case, but it was less "keep this stable for me" and more just an example of what others might do too, now or eventually. Personally I look at binaryen as signing up for a little blinding edge since the Wasm spec is rapidly changing (relatively speaking as far as specs go)

Reply:
I see, let's leave this as is then.